client/data/reader/use-login-window.ts

```diff
@@ -0,0 +1,68 @@
+import config from '@automattic/calypso-config';
+import { createAccountUrl } from 'calypso/lib/paths';
+
+interface UseLoginWindowProps {
+	onLoginSuccess: () => void;
+}
+
+interface UseLoginWindowReturn {
+	login: () => void;
+	createAccount: () => void;
+}
+
+export default function useLoginWindow( {
+	onLoginSuccess,
+}: UseLoginWindowProps ): UseLoginWindowReturn {
+	const isBrowser: boolean = typeof window !== 'undefined';
+	const environment: string = config( 'env_id' );
+	let domain: string = 'wordpress.com';
+	let redirectTo: string = encodeURIComponent(
+		`https://${ domain }/public.api/connect/?action=verify&service=wordpress`
+	);
+	if ( environment === 'development' ) {
+		domain = 'wpcalypso.wordpress.com';
+		redirectTo = encodeURIComponent(
+			`https://${ domain }/public.api/connect/?action=verify&service=wordpress&domain=${ domain }`
+		);
+	}
+
+	const loginURL: string = `https://wordpress.com/log-in?redirect_to=${ redirectTo }`;
+	const createAccountURL: string = `https://wordpress.com${ createAccountUrl( {
+		redirectTo: redirectTo,
+		ref: 'reader-lw',
+	} ) }`;
+	const windowFeatures: string =
+		'status=0,toolbar=0,location=1,menubar=0,directories=0,resizable=1,scrollbars=0,height=980,width=500';
+	const windowName: string = 'CalypsoLogin';
+
+	const waitForLogin = ( event: MessageEvent ) => {
+		if ( event.origin !== `https://${ domain }` ) {
+			return;
+		}
+
+		if ( event?.data?.service === 'wordpress' ) {
+			onLoginSuccess();
```

Comment:
Would it make sense to remove the listener for `waitForLogin` here? (i think you were doing this before)

Im wondering if there any other places we would need to clean up the listener - closing the modal, exiting the login/signup window without completing login, etc. 🤔 . Although im not immediately sure the best way to go about that here.

Reply:
I added the code back that uses setInterval to check if popup closes.